alicloud/data_source_alicloud_emr_instance_types.go

```diff
@@ -0,0 +1,169 @@
+package alicloud
+
+import (
+	"sort"
+	"strings"
+
+	"github.com/aliyun/alibaba-cloud-sdk-go/services/emr"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/terraform-providers/terraform-provider-alicloud/alicloud/connectivity"
+)
+
+func dataSourceAlicloudEmrInstanceTypes() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceAlicloudEmrInstanceTypesRead,
+
+		Schema: map[string]*schema.Schema{
+			"destination_resource": {
+				Type:     schema.TypeString,
+				Required: true,
+				ValidateFunc: validateAllowedStringValue([]string{
+					"Zone",
+					"Network",
+					"InstanceType",
+					"SystemDisk",
+					"DataDisk",
+				}),
+			},
+			"cluster_type": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"instance_charge_type": {
+				Type:     schema.TypeString,
+				Required: true,
+				ValidateFunc: validateAllowedStringValue([]string{
+					"PostPaid",
+					"PrePaid",
+				}),
+			},
+			"output_file": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"ids": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+			},
+			"main_version": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"emr_infos": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"id": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"zone_id": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func dataSourceAlicloudEmrInstanceTypesRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*connectivity.AliyunClient)
+
+	versionRequest := emr.CreateDescribeEmrMainVersionRequest()
+	versionRequest.Scheme = "https"
```

Comment:
I think the line 77 is redundant.
In addition, this data source aims to retrieve instance types. If you want to get mainVersion, you should add a new data source.